File: 백준/Silver/2559. 수열/수열.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int[] num = new int[N];
        int sum = 0;
        
        st = new StringTokenizer(br.readLine());
        for(int i=0; i<N; i++) {
            num[i] = Integer.parseInt(st.nextToken());
            if(i < M) {
                sum += num[i];
            }
        }
        
        int answer = sum;        
        int start = 0, end = M;
        
        while(end < N) {
            sum += num[end++];
            sum -= num[start++];
            answer = Math.max(answer, sum);
        }
        
        System.out.println(answer);
    }
}
    

File: 백준/Silver/2559. 수열/수열.kt
import java.util.StringTokenizer
import kotlin.math.max

//2 2
//3 5
//2 9
fun main() = with(System.`in`.bufferedReader()) {
    var st = StringTokenizer(readLine())
    val N = st.nextToken().toInt()
    val K = st.nextToken().toInt()
    val temp = IntArray(N)
    var sum = 0

    st = StringTokenizer(readLine())
    repeat(temp.size) {
        temp[it] = st.nextToken().toInt()
        if(it < K) {
            sum += temp[it]
        }
    }

    var start = 0
    var end = K
    var answer = sum
    while(end < N) {
        sum -= temp[start++]
        sum += temp[end++]

        answer = max(answer, sum)
    }

    print(answer)
}